cmake/modules/dts.cmake

```diff
@@ -5,7 +5,7 @@ include_guard(GLOBAL)
 include(extensions)
 include(python)
 include(boards)
-include(generic_toolchain)
+find_package(HostTools)
```

Comment:
any reason we are doing `find_package(HostTools)` in `dts.cmake`?

Also why are we handling the host tools as one big package? Should there not be individual packages for: gperf, openocd, bossac, ...

Reply:
because `dts.cmake` requires a generic toolchain (C compiler) which is provided by host tools.

> Also why are we handling the host tools as one big package? 

I did consider if it should be split, but that would result in one of following:
1. Each place someone needs one of the tools, they must now call `FindPackage(<tool>)` themselves first, for example in those places:
    https://github.com/zephyrproject-rtos/zephyr/blob/0cd311c4a1ebad126bdba7858a279c2bb00f7b04/CMakeLists.txt#L1013
    https://github.com/zephyrproject-rtos/zephyr/blob/0cd311c4a1ebad126bdba7858a279c2bb00f7b04/cmake/flash/CMakeLists.txt#L58
2. Create a single module which load all needed and available tools in a single location as general builds system configuration.

Going for 1. would soon become messy, whereas 2. is the solution we have today and this PR just tries to cleanup and places more complex handling like `DTC` detection in dedicated module.

Having 2. ensures that host tools needed for Zephyr development can be loaded using a single code line seen from the caller, not having to consider when a new tool is used.

I will be looking into the generic compiler later, so that hopefully `FindHostTools.cmake` purely becomes something like this in future:
```
find_program(<programA> <a>) # For simple program lookup
find_program(<programB> <b>)

find_packagage(<programC> <version>) # More complex lookup where version and/or other complex handling is needed.
```

It also follows the principle of finding a collection of tools, like: `find_package(UnixCommand)`
Ref: https://cmake.org/cmake/help/latest/module/FindUnixCommands.html